core/src/main/java/org/neo4j/graphalgo/core/utils/ParallelExporter.java

```diff
@@ -1,33 +1,118 @@
 package org.neo4j.graphalgo.core.utils;
 
-import org.neo4j.graphalgo.api.BatchNodeIterable;
+import org.neo4j.graphalgo.api.IdMapping;
+import org.neo4j.graphdb.Transaction;
+import org.neo4j.kernel.api.DataWriteOperations;
+import org.neo4j.kernel.api.Statement;
+import org.neo4j.kernel.api.exceptions.KernelException;
 import org.neo4j.kernel.internal.GraphDatabaseAPI;
+import org.neo4j.logging.Log;
 
+import java.util.ArrayList;
+import java.util.Objects;
 import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicInteger;
 
-public abstract class ParallelExporter<T> extends Exporter<T> {
+/**
+ * @author mknblch
+ */
+public abstract class ParallelExporter<T> extends AbstractExporter<T> {
 
-    private final ExecutorService executor;
+    public static final int DEFAULT_CONCURRENCY = Runtime.getRuntime().availableProcessors() * 2;
```

Comment:
we should move this into some central place, 
also make concurrency configurable